File: DSA/linked_list/linkedListValues.js
class Node {
	constructor(val) {
		this.val = val
		this.next = null
	}
}

const a = new Node('A')
const b = new Node('B')
const c = new Node('C')
const d = new Node('D')

a.next = b
b.next = c
c.next = d

const getValues = (node) => {
	const res = []

	while (node != null) {
		res.push(node.val)
		node = node.next
	}
	return res
}

const getValuesRec = (node) => {
	const res = []
	fillValues(node, res)
	return res
}

const fillValues = (node, res) => {
	if (node === null) return
	res.push(node.val)
	fillValues(node.next, res)
}

console.log('getting values')
console.log(getValues(a))
console.log('getting values recursively')
console.log(getValuesRec(a))

File: DSA/data_structures/linked_list/linked_list_values.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

a = Node('A')
b = Node('B')
c = Node('C')
d = Node('D')

a.next = b
b.next = c
c.next = d

def get_values(node):
    res = []
    while node:
        res.append(node.val)
        node = node.next
    return res

def get_values_rec(node):
    res = []
    fill_values(node, res)
    return res

def fill_values(node, res):
    if node == None:
        return 
    res.append(node.val)
    fill_values(node.next, res)

if __name__ == '__main__':
    print('Getting values')
    print(get_values(a))
    print('Getting values recursively')
    print(get_values_rec(a))
        
